vtex-sites/nextjs.store
# feat: Fire `view_cart` event when the minicart is opened

## What's the purpose of this pull request?

Fire a `view_cart` Analytics event when the minicart is opened.

## How to test it?

1. Visit https://preview-247--nextjs.preview.vtex.app/ and paste the following snippet on the console to log events pushed to the data layer:
```js
window.dataLayerHistory=[];const originalPush=dataLayer.push;dataLayer.push=(data)=>{console.log(data);dataLayerHistory.push(data);originalPush(data)}
```
2. Open the mini cart and you should see the new event on the console.

Empty Cart|Cart with an item
-|-
![CleanShot 2022-09-15 at 12 36 00](https://user-images.githubusercontent.com/381395/190446496-712421fc-230c-4e66-a214-83306225f4a5.png)|![CleanShot 2022-09-15 at 12 35 51](https://user-images.githubusercontent.com/381395/190446509-cabd8562-4fb3-4669-b487-19629b227fe1.png)

I also tested the event being fired on GTM:
![CleanShot 2022-09-15 at 10 48 17](https://user-images.githubusercontent.com/381395/190446872-541bad0c-79f5-40d5-b6d8-f45d4988a72b.png)

## References

- https://developers.google.com/analytics/devguides/collection/ga4/reference/events#view_cart
- https://vtex-dev.atlassian.net/browse/FS-453
